frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ISyncToParentStatusDependentAction.java

```diff
@@ -0,0 +1,103 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+import static io.vavr.API.$;
+import static io.vavr.API.Case;
+import static io.vavr.API.Match;
+import static java.text.MessageFormat.format;
+import static org.checkerframework.checker.i18nformatter.qual.I18nConversionCategory.GENERAL;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.Presentation;
+import io.vavr.collection.List;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.i18nformatter.qual.I18nFormat;
+
+import com.virtuslab.gitmachete.backend.api.SyncToParentStatus;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeyGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.defs.ActionPlaces;
+
+public interface ISyncToParentStatusDependentAction extends IBranchNameProvider, IExpectsKeyGitMacheteRepository {
+  @I18nFormat({})
+  String getActionName();
+
+  @I18nFormat({})
+  String getDescriptionActionName();
+
+  @I18nFormat({GENERAL, GENERAL})
+  String getEnabledDescriptionFormat();
+
+  String getCurrentBranchText();
```

Comment:
Hmmm it's a bit unclear what this method is supposed to return (whether it's just current branch name, wrapper for this name, or sth else)